### PR TITLE
DA-4560: Add new publisher

### PR DESCRIPTION
**Pull request by linonymous**
- add new publisher
- update UserIdentity to UserIdentityObjectBase

Signed-off-by: linonymous <iamlinonymous@gmail.com>

### CODE REVIEW

**Review by lukaszgryglicki — APPROVED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -2,13 +2,16 @@ module github.com/LF-Engineering/insights-datasource-git
 
 go 1.15
 
+replace github.com/LF-Engineering/lfx-event-schema => /Users/swapnilwalke/Documents/go-workspace/LF-Engineering/lfx-event-schema
```
Comment (lukaszgryglicki) [inline go.mod:5]: This won't work for me locally, right? Why do we need this?

Comment (linonymous) [reply go.mod:5]: yes, this wont work. This was done for local testing as the changes I have added in this pull request are dependent upon the https://github.com/LF-Engineering/lfx-event-schema/pull/62 request that is yet to be merged. So to test the changes had to do it. Will need to update this as mentioned pull request gets merged. Its not needed in deployment.

**Comment by lukaszgryglicki** [comment]
That's OK, but when `lfx-event-schema` PR is merged, please remove this.

**Pull request merged**